src/python/pants/backend/go/lint/gofmt/rules.py

```diff
@@ -38,11 +39,30 @@ class GofmtRequest(FmtTargetsRequest):
     partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
 
 
+class GoFmtUnsupportedArgsPassedError(Exception):
+    pass
+
+
+SupportedGoFmtArgs = FrozenOrderedSet(("-e", "-r", "-s"))
```

Comment:
Why only support some args? Can you add a comment explaining?

Reply:
Some options like `-w` are not style related and are used by the rule code to change how `gofmt` operates. This type of option should not be passed through or else it may interfere with the rule.